### PR TITLE
Deprecating the trellis rplot module (GH3445)

### LINKED ISSUE #3445: Deprecating the trellis rplot module
Original title: rplot.rst needs cleaning up

----

http://pandas.pydata.org/pandas-docs/dev/rplot.html

plot style, dimensions, grids, legend.

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #3445 

---

Start for #3445. Then the current examples should get a seaborn or ggplot alternative.

Questions: 
- explicitely refer to seaborn and ggplot as alternatives? (as there may be other packages that could feel neglected that way?)
- I now raise the warning on import of the module (`import pandas.tools.rplot as rplot`). The only thing is that the warning will be raised in the tests this way

### CODE REVIEW

**Comment by TomAugspurger** [comment]
> @TomAugspurger Did you already have some examples of how to do it in seaborn? I vaguely can remember you posted it once in an issue? 

@jorisvandenbossche I redid all the examples using seaborn. A quick search didn't turn it up, but I'll dig for it later.

**Comment by tacaswell** [comment]
There are ways to both suppress and check for warnings in the tests, we have a couple of examples of it in mpl's test suite.

**Comment by shoyer** [comment]
I'm not sure ggplot is ready for prime time. I would probably only recommend Seaborn at this point.

**Comment by jorisvandenbossche** [comment]
@shoyer yes, I could agree, as I noticed while trying to convert some of the examples to gpplot this was not easy (some of the functionality of rplot is not yet included in ggplot). I will update the PR

**Comment by JanSchulz** [comment]
I agree too: ggplot currently doesn't handle categorical data very well if faceting is used (actually that was my motivation for working on categorical data in pandas :-) ).

**Comment by jorisvandenbossche** [comment]
OK, I will remove ggplot for now from the explanation, and if @TomAugspurger finds his seaborn examples, we can add that. But @JanSchulz, if you want to add some ggplot examples when there is some progress, certainly do!

**Comment by jreback** [comment]
@jorisvandenbossche @TomAugspurger let's do this for 0.16.0 (pls add a refernce in #6581) as well

**Comment by jorisvandenbossche** [comment]
I made some examples of how to do it with seaborn, for each example in de rplot docs now: http://nbviewer.ipython.org/gist/jorisvandenbossche/103876bff4a5f42c9933#

@TomAugspurger Can you have a looks if this looks good? Or if there are better ways? (only for the last example (4 variables) I did not found a fully equivalent seaborn-way of doing it)

@jreback For the actual PR, I assume adding seaborn as a doc dependency is not that handy, so I better include all the figures as images?

**Comment by TomAugspurger** [comment]
Joris, I remember talking to Michael about that last plot and it was agreed
that seaborn didn't have a way to do that plot directly. It was also agreed
that this is a good thing, since that chart is a mess :) Thanks for redoing
this

**Comment by jorisvandenbossche** [comment]
I fully agree about that mess :-)

**Comment by jorisvandenbossche** [comment]
The rest is looking OK?

File: doc/source/visualization.rst [hunk 1]
```diff
@@ -1779,38 +1848,12 @@ If the first grouping attribute is not specified the plots will be arranged in a
...
-   ScaleShape(column)
-
-Map the shape of the object to attribute value. The attribute has to be categorical.
+This can also be done in ``seaborn``, at least for 3 variables:
 
-::
+.. code-block:: python
 
-   ScaleRandomColour(column)
+    g = sns.FacetGrid(tips_data, row="sex", col="smoker", hue="day")
```
Comment (shoyer) [inline doc/source/visualization.rst]: you might add `hue_kws={"marker": ["^", "v", "s", "*"]}` if you like the symbols.... or not :).

**Comment by shoyer** [comment]
This looks great to me!

File: doc/source/visualization.rst (hunk 1 above)
Comment (jorisvandenbossche) [inline doc/source/visualization.rst]: Are the different symbols then additional to the different colors? Or in place of?

Comment (shoyer) [inline doc/source/visualization.rst]: The symbols are matched with colors 1 to 1. See the Seaborn docs on plotting on data aware grids for details.

**Comment by jreback** [comment]
@jorisvandenbossche I think just need a release not in the deprecations sections

**Comment by jorisvandenbossche** [comment]
@jreback ah yes!

**Pull request merged**